core/src/main/scala/kafka/server/ConfigHandler.scala

```diff
@@ -70,10 +70,19 @@ class TopicConfigHandler(private val replicaManager: ReplicaManager,
     val logs = logManager.logsByTopic(topic)
     val wasRemoteLogEnabledBeforeUpdate = logs.exists(_.remoteLogEnabled())
 
+    maybeFailIfDisablingRemoteStorage(props, wasRemoteLogEnabledBeforeUpdate)
```

Comment:
Even though this exception is triggered, for some reason the config is still updated; so I decided to trigger the exception earlier in the process on AdminZkClient